aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3062,7 +3063,9 @@ Tensor& linalg_svdvals_out(const Tensor& A, Tensor & S) {
 }
 
 Tensor linalg_svdvals(const Tensor& A) {
-  const bool A_requires_grad = (at::GradMode::is_enabled() && A.requires_grad());
+  const bool A_requires_grad = (at::GradMode::is_enabled() && A.requires_grad())
+                               || A._fw_grad(/*level */ 0).defined()
+                               || isTensorSubclassLike(A);
```

Comment:
I was wondering if `isTensorSubclassLike(A)` shouldn't be:
```
isTensorSubclassLike(A) && at::GradMode::is_enabled()
```
?

Reply:
good point, let me test that

Reply:
Actually, I'm leaving it as it is, because it might be the case we're wrapping somehing in forward mode, where we can have `GradMode` disabled

Reply:
As noted by @albanD offline, the comment above is right, we cannot use `GradMode` here.